tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -1480,11 +3042,82 @@ static function() {
 		);
 
 		// The non-default script should end concatenation and maintain order.
-		$ver       = get_bloginfo( 'version' );
-		$expected  = "<script type='text/javascript' src='/wp-admin/load-scripts.php?c=0&amp;load%5Bchunk_0%5D=jquery-core&amp;ver={$ver}'></script>\n";
+		$expected  = "<script type='text/javascript' src='/wp-admin/load-scripts.php?c=0&amp;load%5Bchunk_0%5D=jquery-core&amp;ver={$wp_version}'></script>\n";
 		$expected .= "<script type='text/javascript' src='/plugins/wp-i18n.js' id='wp-i18n-js'></script>\n";
 		$expected .= "<script type='text/javascript' src='/default/common.js' id='common-js'></script>\n";
 
 		$this->assertSame( $expected, $print_scripts );
 	}
+
+	/**
+	 * Gets the script tag for the delayed inline script loader.
+	 *
+	 * @return string Script tag.
+	 */
+	protected function get_delayed_inline_script_loader_script_tag() {
+		/*
+		 * Ensure the built delayed inline script loader file exists
+		 * when the test suite is run from the 'src' directory.
+		 *
+		 * Note this should no longer be needed as of https://core.trac.wordpress.org/ticket/57844.
+		 */
+		$build_path = ABSPATH . WPINC . '/js/wp-delayed-inline-script-loader' . wp_scripts_get_suffix() . '.js';
+
+		if ( ! file_exists( $build_path ) ) {
+			$src_path = ABSPATH . 'js/_enqueues/lib/delayed-inline-script-loader.js';
+
+			$file_contents = file_get_contents( $src_path );
+
+			self::touch( $build_path );
+			file_put_contents(
+				$build_path,
+				$file_contents
+			);
+		}
+
+		return wp_get_inline_script_tag(
+			file_get_contents( $build_path ),
+			array( 'id' => 'wp-delayed-inline-script-loader' )
+		);
+	}
+
+	/**
+	 * Parse an HTML markup fragment.
+	 *
+	 * @param string $markup Markup.
+	 * @return DOMElement Body element wrapping supplied markup fragment.
+	 */
+	protected function parse_markup_fragment( $markup ) {
+		$dom = new DOMDocument();
+		$dom->loadHTML(
+			"<!DOCTYPE html><html><head><meta charset=utf8></head><body>{$markup}</body></html>"
+		);
+
+		/** @var DOMElement $body */
+		$body = $dom->getElementsByTagName( 'body' )->item( 0 );
+
+		// Trim whitespace nodes added before/after which can be added when parsing.
+		foreach ( array( $body->firstChild, $body->lastChild ) as $node ) {
+			if ( $node instanceof DOMText && '' === trim( $node->data ) ) {
+				$body->removeChild( $node );
+			}
+		}
+
+		return $body;
+	}
+
```

Comment:
After https://github.com/WordPress/wordpress-develop/commit/559e6cecf49505ecc3a21fbff647fdc5c3f892a3, I want to make sure that `wp_dequeue_script` and `wp_deregister_script` and see what happens. Can we add some unit tests, with script that have defer / async and dependencies and deregister / dequeue and see what happens?

Reply:
I don't believe this is needed because the logic for rendering is contained within `do_items()` so after any `wp_dequeue_script()`/`wp_deregister_script()` calls have been made. So I don't see how they could cause problems.